interop-twitter/jvm/src/main/scala/scalaz/zio/interop/twitter.scala

```diff
@@ -0,0 +1,18 @@
+package scalaz.zio.interop
+
+import com.twitter.util.{ Future, Return, Throw }
+import scalaz.zio.Task
+
+package object twitter {
+  implicit class TaskObjOps(private val obj: Task.type) extends AnyVal {
+    final def fromTwitterFuture[A](future: => Future[A]): Task[A] =
+      Task.effectAsync { cb =>
```

Comment:
You will also need to handle interruption from the outside with `future.raise`

Reply:
Any suggestions on the approach? From what I see, interrupt handler is available only for `Promise`, and raising is propagated to producers.

Reply:
I've made some change [here](https://github.com/yaroot/scalaz-zio/commits/twitter-future)

Interruption handler in promise is where you need to accept downstream cancellation, here you need to signal cancellation by using `future.raise`.

Besides that `fromTwitterFuture` should really accept `Task[Future[A]]` (similar to the scala version `fromFuture`).

Just my 2 cents

Reply:
As @yaroot did 😄

Reply:
@yaroot `fromFuture` doesn't accept `Task`, as far as I can see, it accepts `ExecutionContext => scala.concurrent.Future[A]`. I'll take a look at the solution you suggested, or, I can retract the PR and let you submit the solution, if it's the preferred approach :).